### ISSUE TITLE
Color keying and PNG transparency

### ISSUE THREAD

**Issue by acowley** [jxv/dino-rush#3]
When I built and ran the game on my macOS machine with nix, the color keying for transparency didn't work, so I had a lot of pink/purple on the screen. 

I found this [question](https://stackoverflow.com/questions/36666775/converting-sdl2-color-keyed-surface-to-texture-no-transparent-pixels) with a comment suggesting that having an alpha channel in the PNG file could fight the color keying. I used imagemagick's `convert` utility to remove the alpha channel from each PNG, and -- lo! -- it worked.

Not sure how widespread the issue might be, but it seems like it couldn't hurt to remove the alpha channel if it's not being used. I can PR the no-alpha PNGs if it's helpful.

**Comment by jxv** [comment]
@acowley Hmm... What's your architecture? This seems sort of familiar. I may have I've encountered this issue before which I think was a little-endian vs big-endian issue. This was how it was solved with SDL 1. https://github.com/jxv/gridland/blob/master/src/GridLand/SDL.hs#L49

If not, digging into how the imagemagick's `convert` utility works would shed some light.

**Comment by acowley** [comment]
I don't think `convert` is doing anything interesting. My guess would be along the lines of the comments on that link I included: it either has to do with some default OpenGL texture format, and that may end up being informed by the windowing system, or it's that image loading on some platforms discards alpha from PNGs while on others it is retained. If the image data from the PNG has no alpha channel, SDL uses the color key to synthesize it when filling the texture (which is what we want).

In any case, the original PNGs do have alpha channels (I don't know if you have a macOS machine or not, but I checked this by opening them in Preview.app and looking at the More Info tab of the Inspector window), which they probably don't need. Some image editing applications save a fully opaque alpha channel by default, which is what I'd guess has happened here.

It's not a big deal, the game works and is fun to play even when the sprite masking isn't working. Call it collision box visibility mode 😜
